platform/core.windows/src/org/netbeans/core/windows/services/DialogDisplayerImpl.java

```diff
@@ -227,9 +239,21 @@ public void showDialog () {
             Window parent = noParent ? null : findDialogParent();
 
             if (descriptor instanceof DialogDescriptor) {
-                presenter = new NbDialog((DialogDescriptor) descriptor, parent);
+                if (parent instanceof Dialog) {
+                    presenter = new NbDialog((DialogDescriptor) descriptor, (Dialog) parent);
+                } else if (parent instanceof Frame) {
+                    presenter = new NbDialog((DialogDescriptor) descriptor, (Frame) parent);
+                } else {
+                    presenter = new NbDialog((DialogDescriptor) descriptor, (Frame) null);
+                }
```

Comment:
I had that at one point where `parent` could still be a non-null `Window`. That should now not be possible, but nothing wrong with a little caution rather than a ClassCastException :smile:

Reply:
> change from Frame to Window

I was talking about the tweak you made to this PR around an hour ago
https://github.com/apache/netbeans/compare/9d72942c66f78a3efb3419a164eef17cb552da28..6468be111f64ea255293bcfd98f728c96e55f609

Reply:
Well, I first changed it to pass the test, then realised that the test had been changed.  Prefer reverting to the old test unless there's a reason not to.

Reply:
Cool. That's the crux. I was wondering if there was a reason to prefer one over the other. Guess not; just another corner case and as long as it doesn't crash or lockup...

Reply:
There was a slight preference from me to use the NB18 and earlier behaviour for "parentless" dialogs, which should only happen rarely (or at startup) anyway.